File: POO - Python & Java/practica-1/src/gestorAplicacion/Paciente.java
package gestorAplicacion;

import java.io.Serializable;
import java.util.ArrayList;

public class Paciente extends Persona implements Serializable{
	
	/*	INSTANCE METHODS	*/
	private static final long serialVersionUID = 1L;
	private Cita cita;
	private int turno = -1;
	private ArrayList<Sintomas> sintomas = new ArrayList<Sintomas>();
	private ArrayList<Especialidad> tipoProblemasMedicos = new ArrayList<Especialidad>();
	private static int numeroPacientes;
	
	
	/*	CONSTRUCTORS	*/
	public Paciente(String nombre, int cedula) {
		this(nombre, cedula, false, null, null);
	}
	public Paciente(String nombre, int cedula, boolean poblacionDeRiesgo) {
		this(nombre, cedula, poblacionDeRiesgo, null, null);
	}
	public Paciente(String nombre, int cedula, boolean poblacionDeRiesgo,Sintomas sintoma) {
		super(nombre, cedula, poblacionDeRiesgo);
		this.sintomas.add(sintoma);
		Paciente.numeroPacientes++;
	}
	
	public Paciente(String nombre, int cedula, Sintomas sintoma) {
		this(nombre, cedula, false, new Sintomas[]{sintoma}, -1);
	}

	public Paciente(String nombre, int cedula, boolean poblacionDeRiesgo, Sintomas[] sintomas, int turno) {
		super(nombre, cedula, poblacionDeRiesgo, 0);
		Paciente.numeroPacientes++;
		
		for (int i = 0; i < sintomas.length; i++) {
			this.sintomas.add(sintomas[i]);
		}
		
		this.turno = turno;
		this.diagnosticar();
	}
	
	
	public Paciente(String nombre, int cedula, boolean poblacionDeRiesgo, Sintomas[] sintomas, Cita cita) {
		super(nombre, cedula, poblacionDeRiesgo, 0);
		
		if(sintomas != null) {
			for (int i = 0; i < sintomas.length; i++) {
				this.sintomas.add(sintomas[i]);
			}
		}
		
		this.cita = cita;
	}
	
	/*
	 * diagnosticar: El sistema le asigna a que especialista busca el paciente
	 */
	public void diagnosticar() {
		for (Sintomas sintoma : this.getSintomas()) {
			if(sintoma == Sintomas.DOLORPECHO) {
				this.addTipoProblemasMedicos(Especialidad.CARDIOLOGIA);
			} else if(sintoma == Sintomas.DEMENCIA) {
				this.addTipoProblemasMedicos(Especialidad.NEUROCIRUJANO);
			} else if(sintoma == Sintomas.PIELRARA) {
				this.addTipoProblemasMedicos(Especialidad.DERMATOLOGIA);
			}
		}
		
		if(this.getTipoProblemasMedicos().size() == 0) {
			this.addTipoProblemasMedicos(Especialidad.GENERAL);
		}
	}
	/*
	 * 
	 */
	public String toString() {
		return "Paciente " + nombre + " identificado con documento " + cedula;
	}
	
	public boolean infectar() {
		boolean retorno = super.infectar();
		addSintomas(Sintomas.DEMENCIA);
		return retorno;
	}
	
	/*	GETTERS & SETTERS	*/
	public ArrayList<Sintomas> getSintomas() { return this.sintomas; }
	public void addSintomas(Sintomas[] sintomas) {
		for (int i = 0; i < sintomas.length; i++) {
			this.sintomas.add(sintomas[i]);
		}
	}
	public void addSintomas(Sintomas sintoma) {
		this.sintomas.add(sintoma);
	}

	public String getNombre() { return super.getNombre();} 

	public Cita getCita() { return this.cita; }
	public void setCita(Cita cita) { this.cita = cita; }
	
	public int getTurno() { return this.turno; }
	public void setTurno(int turno) { this.turno = turno; }
	
	public ArrayList<Especialidad> getTipoProblemasMedicos() { return this.tipoProblemasMedicos; }
	public void setTipoProblemasMedicos(ArrayList<Especialidad> tipoProblemasMedicos) { this.tipoProblemasMedicos = tipoProblemasMedicos; }
	public void addTipoProblemasMedicos(Especialidad tipoProblemasMedicos) { this.tipoProblemasMedicos.add(tipoProblemasMedicos); }
}

File: POO - Python & Java/practica-2/gestorAplicacion/Paciente.py
from gestorAplicacion.Persona import Persona
from gestorAplicacion.Especialidad import Especialidad
from gestorAplicacion.Sintomas import Sintomas

class Paciente(Persona):

    def _initialize_instance_fields(self):
        #instance fields found by Java to Python Converter:
        self._cita = None
        self._turno = -1
        self._sintomas = []
        self._tipoProblemasMedicos = []

    _numeroPacientes = 0

    def __init__(self, nombre, cedula, poblacionDeRiesgo, sintomas, turno=-1):
        self._initialize_instance_fields()

        super().__init__(nombre, cedula, poblacionDeRiesgo, 0)
        Paciente._numeroPacientes += 1

        i = 0
        while i < len(sintomas):
            self._sintomas.append(sintomas[i])
            i += 1

        self._turno = turno
        self.diagnosticar()

    #	
    #	 * diagnosticar: El sistema le asigna a que especialista busca el paciente
    #	 
    def diagnosticar(self):
        for sintoma in self.getSintomas():
            if sintoma == Sintomas.DOLORPECHO:
                self.addTipoProblemasMedicos(Especialidad.CARDIOLOGIA)
            elif sintoma == Sintomas.DEMENCIA:
                self.addTipoProblemasMedicos(Especialidad.NEUROCIRUJANO)
            elif sintoma == Sintomas.PIELRARA:
                self.addTipoProblemasMedicos(Especialidad.DERMATOLOGIA)

        if len(self.getTipoProblemasMedicos()) == 0:
            self.addTipoProblemasMedicos(Especialidad.GENERAL)
    #	
    #	 * 
    #	 
    def toString(self):
        return "Paciente " + self.nombre + " identificado con documento " + str(self.cedula)

    def infectar(self):
        retorno = super().infectar()
        self.addSintomas(Sintomas.DEMENCIA)
        return retorno

    #	GETTERS & SETTERS	
    def getSintomas(self):
        return self._sintomas
    def addSintomas(self, sintomas):
        i = 0
        while i < len(sintomas):
            self._sintomas.append(sintomas[i])
            i += 1
    def addSintomas(self, sintoma):
        self._sintomas.append(sintoma)

    def getNombre(self):
        return super().getNombre()

    def getCita(self):
        return self._cita
    def setCita(self, cita):
        self._cita = cita

    def getTurno(self):
        return self._turno
    def setTurno(self, turno):
        self._turno = turno

    def getTipoProblemasMedicos(self):
        return self._tipoProblemasMedicos
    def setTipoProblemasMedicos(self, tipoProblemasMedicos):
        self._tipoProblemasMedicos = tipoProblemasMedicos
    def addTipoProblemasMedicos(self, tipoProblemasMedicos):
        self._tipoProblemasMedicos.append(tipoProblemasMedicos)
